python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py

```diff
@@ -41,24 +48,78 @@ class LightGBMTrainer(Trainer):
             dataset. All datasets will be transformed by the ``preprocessor`` if
             one is provided. All non-training datasets will be used as separate
             validation sets, each reporting a separate metric.
-        lightgbm_config: LightGBM training parameters. Refer to
-            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
-/latest/Parameters.html>`_ for a list of possible parameters.
+        lightgbm_config: LightGBM training parameters passed to ``lightgbm.train()``.
+            Refer to `LightGBM documentation <https://lightgbm.readthedocs.io>`_
+            for a list of possible parameters. Must contain a "params" key.
+            You can also pass a "dmatrix_params" key, which should be a dict
+            of ``dataset name:dict of kwargs`` passed to respective
+            :class:`xgboost_ray.RayDMatrix` initializations.
         scaling_config: Configuration for how to scale data parallel training.
         run_config: Configuration for the execution of the training run.
         preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
             provided datasets.
         resume_from_checkpoint: A checkpoint to resume training from.
     """
 
+    _use_lightgbm: bool = True
+
     def __init__(
         self,
         label_column: str,
         datasets: Dict[str, GenDataset],
-        lightgbm_config: Optional[Dict[str, Any]] = None,
+        lightgbm_config: Dict[str, Any],
         scaling_config: Optional[ScalingConfig] = None,
         run_config: Optional[RunConfig] = None,
         preprocessor: Optional[Preprocessor] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
     ):
-        raise NotImplementedError
+        if TRAIN_DATASET_KEY not in datasets:
```

Comment:
So it seems like the logic is almost exactly the same between the 2 Trainers with the only differences being the arg names, the training library, name, and the error messages. We don't have to do this now, but I'm wondering if it would be possible to all the repeat code can just be in the superclass, and then the subclasses is able to configure this with just a super call, like we have now for TorchTrainer and TFTrainer.